code/game/objects/structures/stool_bed_chair_nest/bed.dm

```diff
@@ -127,6 +137,26 @@
 	if(M.pulledby)
 		M.pulledby.start_pulling(src)
 
+/obj/structure/bed/unlock_atom(var/atom/movable/AM)
+	if(glued)
+		return FALSE
+	return ..()
+
+/obj/structure/bed/Destroy()
+	if(glued && is_locking(lock_type))
+		glued = FALSE // So that unlock_atom called in /atom/movable/Destroy can succeed
+		var/mob/living/carbon/human/locked = get_locked(lock_type)[1]
+		if(istype(locked) && locked.op_stage.butt != 4) // does the target have an ass
+			var/obj/item/clothing/head/butt/ass = new(locked.loc)
+			ass.transfer_buttdentity(locked)
+			locked.op_stage.butt = 4 //No having two butts.
```

Comment:
I'm pretty sure I made a define about this one

Reply:
the whole ass removal thing should be a proc in itself, too